polkadot/runtime/common/src/paras_sudo_wrapper.rs

```diff
@@ -80,11 +80,14 @@ pub mod pallet {
 			genesis: ParaGenesisArgs,
 		) -> DispatchResult {
 			ensure_root(origin)?;
+
+			if genesis.para_kind == ParaKind::Parachain {
+				T::AssignCoretime::assign_coretime(id)?;
+			}
+
 			polkadot_runtime_parachains::schedule_para_initialize::<T>(id, genesis)
 				.map_err(|_| Error::<T>::ParaAlreadyExists)?;
 
-			T::AssignCoretime::assign_coretime(id)?;
```

Comment:
The behavior changes: previously, if scheduling para initialization failed, we wouldn't assign coretime, and now, we first assign it and do not unassign if scheduling fails. Are we okay with that?

Reply:
Good point. I'll fix it.